### PR TITLE
Pokemon Emerald: Define additional item and location groups.

### PR DESCRIPTION
**Pull request by Tsukino-uwu**
item/location groups for doing example bulk local/non local filters
instead of having to type out 42 specific berrie names etc, if you dislike berries
this makes it so you can just write "Berries" or even "Berry Trees", 
and also full routes/areas with just "Route 123" or "Victory Road"
even "Team Aqua" or "Team Magma" for all areas where you fight/get items related to them

Only having "Badges", "Trick House" & "Everything" felt a bit limiting/unspecific

spent like 5pm~ - 4am ish on this, took a while to sort/organize stuffs but at least I could start from/with the datapackage

## What is this fixing or adding?
**Location groups:**

Safari Zone, Abandoned Ship, SS Tidal, All Routes, All Cities, All Towns, All Ships, All Caves, All Mountains, Aqua Hideout, Magma Hideout

Team Aqua # all aqua locations
Team Magma

Petalburg Woods, Oceanic Museum, Space Center

All cities,towns,caves,mountains separately 

Trainers #All trainers including grunts, gym leaders etc

Npc Gifts # All items received from npcs

Berry Trees # all tree locations

Visible Items # Includes all above ground visible items

Hidden Items # Includes all hidden items

Hidden UW Items # Only includes underwater items

**Item groups:** 

Berries #all 42 berries

Battle Items # X items etc

Consumables 

Vitamins 

Key Items 

Bikes

Rods 

Keys #Just keys, not key items

Tickets

Fossils

Poke Balls 

Mails 

Evolution Stones 

TMs # also comments on the ones that are received by npcs

Held items # does not include berries or mails

Filler Items # for sorting etc items that don't fit anywhere else

## How was this tested?
Haven't been tested, but shouldn't break anything

### CODE REVIEW

File: worlds/pokemon_emerald/items.py [hunk 1]
```diff
@@ -67,6 +67,330 @@ def create_item_label_to_code_map() -> Dict[str, int]:
     "HM06": {"HM06 Rock Smash"},
     "HM07": {"HM07 Waterfall"},
     "HM08": {"HM08 Dive"},
+    },
```
Comment (ScipioWright) [inline worlds/pokemon_emerald/items.py:70]:
```suggestion

```
Doesn't work, delete this to make it work

**Comment by Zunawe** [comment]
I wish you would have mentioned to me first that you'd like to do this, because there are better ways that I've had ideas for but haven't had the time/energy to implement yet.

Item groups and some location groups are basically duplicating the `tag` property in `worlds\pokemon_emerald\data\locations.json` and `worlds\pokemon_emerald\data\items.json`. It would be better to use that to automatically create groups instead of listing them all out again.

For many of the "geographical" location groups, I'd prefer to instead assign each map a location group. Every location already gets a parent region, and every region already gets a parent map. So you'd only be mapping maps to location groups, and all locations contained within the map would be picked up automatically.

And then you could abstract it one final time to create location groups by concatenating other location groups if you really want big overarching groups like "Dungeons" or something.

There are some that I also don't agree with, like `All Mountains` or `Team Magma`, which are without purpose or with far too niche a purpose in my opinion. The only reason I can imagine anyone would use a Team Magma group would be some sort of specific themed yaml. But almost everyone will simply have to scroll past it to get to the more useful ones.

If you want to convert this PR to a method more like I've described, let me know. I'm not super happy with the way I've done "tags" in the first place anyway, so I'll want to consider reworking them a little bit, and I don't want to change that out from under you.

**Comment by Zunawe** [comment]
Forgot to explicitly say

The _reason_ I want to do it all this way is because then I never have to care about updating it again. The current list is pretty manageable because it's so small. But in the future, any time I add, rename, remove, or recategorize locations, I don't want to have to remember to also add it to the relevant location groups. If it's all automatically picked up based on the region/tags, then, well, I already have to do that work anyway no matter what, so it's taken care of for me.

**Comment by Zunawe** [comment]
Closing in favor of #3263

**Pull request closed without merging**